gio/src/list_store.rs

```diff
@@ -43,20 +44,28 @@ impl ListStore {
         }
     }
 
+    /// Dumbest possible example just to test this. Will remove.
+    /// ```
+    /// use glib::prelude::*;
+    /// use gio::ListStore;
+    /// use gio::prelude::*;
+    /// let item1 = ListStore::new(ListStore::static_type());
+    /// let item2 = ListStore::new(ListStore::static_type());
+    /// let list = ListStore::new(ListStore::static_type());
+    /// list.splice(0, 0, &[item1, item2.clone()]);
+    /// assert!(list.item(1) == Some(item2.upcast()))
```

Comment:
Can you put that into a `#[test]` in this module? That way we can make sure it continues to work :)